### PR TITLE
validation: add checks for `L2OutputOracle` parameters

### PR DESCRIPTION
**Pull request by geoknee**
**Description**

* adds (exported) expectations / ground truths to the `superchain` module resolve by superchain target
* Adds a test to query parameters from the `L2OutputOracle` contract for each chain and check them against expectations

**Metadata**

- Towards https://github.com/ethereum-optimism/client-pod/issues/486 (task 3)

### CODE REVIEW

File: validation/l2oo_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func TestL2OOParams(t *testing.T) {
+
+	isExcluded := map[uint64]bool{
+		10:           true, // mainnet/op                      (old version of L2OutputOracle, no submissionInterval method)
```
Comment (geoknee) [inline validation/l2oo_test.go:23]: Notice most chains have to be excluded for this reason. Note also that[ the old contract does expose a `SUBMISSION_INTERVAL` getter. ](https://etherscan.io/address/0xdfe97868233d1aa22e815a266982f2cf17685a27#readProxyContract). I can add a workaround for this, but I wasn't sure if it was worth it (presumably future chains will have the newer version of the contract)?

File: superchain/params.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import "math/big"
+
+type L2OOParams struct {
+	StartingBlockNumber       *big.Int // The number of the first L2 block.
+	StartingTimestamp         *big.Int // The timestamp of the first L2 block.
+	SubmissionInterval        *big.Int // Interval in blocks at which checkpoints must be submitted.
+	L2BlockTime               *big.Int // The time per L2 block, in seconds.
+	FinalizationPeriodSeconds *big.Int // The minimum time (in seconds) that must elapse before a withdrawal can be finalized.
+}
```
Comment (geoknee) [inline superchain/params.go:11]: I'm unsure if this is a definitive list of parameters to check. In https://github.com/ethereum-optimism/client-pod/issues/210#issuecomment-1890339256, only `SubmissionInterval` and `FinalizationPeriodSeconds` are mentioned. I'm fairly sure `L2BlockTime` is important. But maybe not e.g. `StartingTimestamp`?

File: superchain/superchain.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ func (a AddressList) AddressFor(contractName string) (Address, error) {
 		address = a.L1ERC721BridgeProxy
 	case "L1StandardBridgeProxy":
 		address = a.L1StandardBridgeProxy
-	case "L2OutputOrcaleProxy":
+	case "L2OutputOracleProxy":
```
Comment (mslipper) [inline superchain/superchain.go:90]: wow nice catch, surprised this didn't cause any problems before

File: validation/l2oo_test.go (hunk 1 above)
Comment (mslipper) [reply validation/l2oo_test.go:23]: These chains will be upgraded in the future to include that getter, and once FPs launch the L2OO will be replaced altogether. I'm fine with this solution for now.

File: validation/l2oo_test.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	params.StartingBlockNumber, err = retry.Do(ctx, maxAttempts, retry.Exponential(),
+		func() (*big.Int, error) {
+			return l2OO.StartingBlockNumber(&bind.CallOpts{Context: ctx})
+		})
+	if err != nil {
+		return L2OOParams{}, fmt.Errorf("could not get startingBlockNumber: %w", err)
+	}
+	params.StartingTimestamp, err = retry.Do(ctx, maxAttempts, retry.Exponential(),
+		func() (*big.Int, error) {
+			return l2OO.StartingTimestamp(&bind.CallOpts{Context: ctx})
```
Comment (mslipper) [inline validation/l2oo_test.go:131]: [nit] You can reuse the callopts

**Review by mslipper — APPROVED** [review]
LGTM, left one nit

File: superchain/params.go (hunk 2 above)
Comment (geoknee) [reply superchain/params.go]: [f6be092](https://github.com/ethereum-optimism/superchain-registry/pull/79/commits/f6be092a9e3a56672b737fd3e7f62281ad46d33c)

File: superchain/superchain.go (hunk 3 above)
Comment (geoknee) [reply superchain/superchain.go:90]: Yeah I guess this is a downside to this approach (of mapping runtime information to static / type information). Maybe tips the scale back a tiny bit towards using reflection...

File: validation/l2oo_test.go (hunk 4 above)
Comment (geoknee) [reply validation/l2oo_test.go]: [fce979a](https://github.com/ethereum-optimism/superchain-registry/pull/79/commits/fce979a3da3f98b6f90914c5425208b0858481d2)

**Pull request merged**